### ISSUE TITLE
Add read time stats

### ISSUE THREAD

**Issue by mathewthe2** [lrorpilla/jidoujisho#150]
Ever since I've added a [read time tracker](https://github.com/mathewthe2/immersion_reader/issues/4) to my reader app, I've found it helpful to keep my self consistent and accountable.

Having these stats for jidoujisho would be great:
- daily read time / goal
- reading streak
- read time per page (for particular book or globally)
- characters read per minute (for particular book or globally)

The main difficulty for tracking read time is detecting when the user sleeps the app (switches apps / turns off screen) and when they restart the app to read.

```dart
  Future<void> handleAppResume() async {
    startLocalAssetsServer();
    profileProvider?.restartSession();
  }

  Future<void> handleAppSleep() async {
    stopLocalAssetsServer();
    profileProvider?.endSession();
  }

  @override
  void didChangeAppLifecycleState(AppLifecycleState state) {
    super.didChangeAppLifecycleState(state);
    switch (state) {
      case AppLifecycleState.resumed:
        handleAppResume();
        break;
      case AppLifecycleState.inactive:
      case AppLifecycleState.paused:
      case AppLifecycleState.detached:
        handleAppSleep();
        break;
    }
  }
```

**Comment by lrorpilla** [comment]
I've decided in regards to this issue, I might wait on @KamWithK and integrate with [**exSTATic**](https://github.com/KamWithK/exSTATic) at some point in the future. 

I know this is something that people really want, but I think it would be best if I can let other developers do what they do best and I can focus on areas that I have to implement solely on my own -- it would potentially be more feature-rich than any simple solution I work on too.

**Comment by KamWithK** [comment]
Getting stat tracking right is really crucial for immersion learning so I'm super glad to work on providing a way for [exSTATic](https://github.com/KamWithK/exSTATic) to integrate with apps like Jidoujisho! I've got pretty much all the graphs you would want (happy to add more too), so keen to enable more people to use it.

Jidoujisho will definitely be one of the first new integrations (if not the first) with exSTATic, but to be clear the rework will probably take at least a few months (on pause currently due to how busy I am with other stuff). Hope the wait is okay @lrorpilla :)

**Comment by mathewthe2** [comment]
So can everything be run locally on my phone without internet or is this going to be an cloud service or has to be hosted on my computer?

**Comment by mathewthe2** [comment]
> It currently works via an offline webextension

I mean can it be embedded into this app as an offline module?

**Comment by KamWithK** [comment]
> 

Eventually things will be worked out
It will require work on both our ends but it should be doable in due time (not sure if that's the answer you were looking for but it's as good an answer as can be provided by me currently)

**Comment by NielsVandenEynde** [comment]
Any updates on this?

**Comment by KamWithK** [comment]
This hasn't been a priority but I have got auth working, I just need to get a sync server for stats up, confirm it works for exSTATic website and then we can integrate (if jidoujisho is still keen on exSTATic support)

**Comment by ElaxDev** [comment]
Ttu reader now has stats, for the time being Jidoujisho could change to use https://reader.ttsu.app/ so we have stats on the reader

**Comment by brishtibheja** [comment]
yeah we need an update on this. can anyone contact him in TMW's discord? he may not be reading through the issue tracker.

**Comment by lrorpilla** [comment]
Rest assured I am reading all this. I am working on something and it will take time.

**Comment by brishtibheja** [comment]
Hey, will ttu reader be updated or we will need to wait until you finish the new version?

**Comment by brishtibheja** [comment]
@Irorpilla Can you at least release a new previewer version with updated ttu reader? That will get us the stats. The new stuff you're working on might take more time so releasing a new ver. in the recent future will help us.

**Comment by HitogiDev** [comment]
Updates on this?
